src/theia/sfm/global_pose_estimation/robust_rotation_estimator.cc

```diff
@@ -221,19 +183,26 @@ bool RobustRotationEstimator::SolveIRLS() {
     return false;
   }
 
-  VLOG(2) << "Iteration   Error           Delta";
+  VLOG(2) << "Iteration   SqError         Delta";
   const std::string row_format = "  % 4d     % 4.4e     % 4.4e";
 
-  Eigen::ArrayXd errors, weights;
+  ComputeResiduals();
+
+
+  Eigen::ArrayXd weights(kNumEdges * 3);
   Eigen::SparseMatrix<double> at_weight;
   for (int i = 0; i < options_.max_num_irls_iterations; i++) {
-    const Eigen::VectorXd prev_rotation_change = rotation_change_;
-    ComputeRotationError();
 
-    // Compute the weights for each error term.
-    errors =
-        (sparse_matrix_ * rotation_change_ - relative_rotation_error_).array();
-    weights = kSigma / (errors.square() + kSigma * kSigma).square();
+    // Compute the Huber-like weights for each error term.
+    for (int k = 0; k < kNumEdges; ++k) {
+      const double& kSigma = options_.irls_loss_parameter_sigma;
+      double e_sq = tangent_space_residual_.segment<3>(3 * k).squaredNorm();
+      double tmp = e_sq + kSigma * kSigma;
+      double w = kSigma / tmp / tmp;
```

Comment:
const doubles

Also please change the expression to kSigma / (tmp * tmp) I think this is a little more safe for future-proofing